### PR TITLE
now sending/recieving json via go structs instead of plain text

### PR DESCRIPTION
**Pull request by markhaur**
now sending/recieving json via go structs instead of plain text.
We might need to do some tweaks for eradicating duplication of code.

### CODE REVIEW

File: cmd/client/main.go [hunk 1]
```diff
@@ -10,6 +10,14 @@ import (
 	"github.com/gorilla/websocket"
 )
 
+type Message struct {
+	MsgId     int64  `json:"message_id"`
```
Comment (jarri-abidi) [inline cmd/client/main.go]:
```suggestion
	MsgID     int64  `json: "message_id"`
```

File: cmd/client/main.go [hunk 2]
```diff
@@ -24,13 +32,14 @@ func main() {
 	done := make(chan struct{})
 	go func() {
 		defer close(done)
+		var msg Message
 		for {
-			mt, msg, err := conn.ReadMessage()
+			conn.ReadJSON(&msg)
```
Comment (jarri-abidi) [inline cmd/client/main.go]: why not checking errors here?

File: cmd/client/main.go [hunk 3]
```diff
@@ -24,13 +32,14 @@ func main() {
...
 			if err != nil {
 				logger.Log("err", err)
 				return
 			}
-			logger.Log("type", mt, "from", "server", "msg", string(msg))
+			logger.Log("from", "server", "msgId", msg.MsgId, "from", msg.MsgFrom, "To", msg.MsgTo, "content", msg.Content, "createdat", msg.CreatedAt)
```
Comment (jarri-abidi) [inline cmd/client/main.go]:
```suggestion
			logger.Log("from", "server", "msgId", msg.MsgId, "from", msg.MsgFrom, "To", msg.MsgTo, "content", msg.Content, "createdAt", msg.CreatedAt)
```

File: cmd/client/main.go [hunk 4]
```diff
@@ -39,13 +48,21 @@ func main() {
 
 	interrupt := make(chan os.Signal, 1)
 	signal.Notify(interrupt, os.Interrupt)
-
+	var idcounter int64
```
Comment (jarri-abidi) [inline cmd/client/main.go]:
```suggestion
	var idCounter int64
```

File: cmd/server/chat/main.go [hunk 5]
```diff
@@ -8,6 +8,14 @@ import (
 	"github.com/gorilla/websocket"
 )
 
+type Message struct {
+	MsgId     int64  `json:"message_id"`
```
Comment (jarri-abidi) [inline cmd/server/chat/main.go]:
```suggestion
	MsgID     int64  `json:"message_id"`
```

File: cmd/server/chat/main.go [hunk 6]
```diff
@@ -24,15 +32,18 @@ func ws(logger log.Logger) http.HandlerFunc {
 		}
 		defer conn.Close()
 
+		var msg Message
 		for {
-			mt, msg, err := conn.ReadMessage()
+			conn.ReadJSON(&msg)
```
Comment (jarri-abidi) [inline cmd/server/chat/main.go]: err check?

File: cmd/client/main.go (hunk 1 above)
Comment (markhaur) [reply cmd/client/main.go]: ok, from now will follow mixed case

File: cmd/client/main.go (hunk 2 above)
Comment (markhaur) [reply cmd/client/main.go]: lemme do it.

File: cmd/client/main.go (hunk 3 above)
Comment (markhaur) [reply cmd/client/main.go]: cool!

File: cmd/client/main.go (hunk 4 above)
Comment (markhaur) [reply cmd/client/main.go]: cool

File: cmd/server/chat/main.go (hunk 5 above)
Comment (markhaur) [reply cmd/server/chat/main.go]: cool

File: cmd/server/chat/main.go (hunk 6 above)
Comment (markhaur) [reply cmd/server/chat/main.go]: lemme do it

File: cmd/client/main.go [hunk 7]
```diff
@@ -24,13 +32,14 @@ func main() {
...
+		var msg Message
 		for {
-			mt, msg, err := conn.ReadMessage()
+			conn.ReadJSON(&msg)
 			if err != nil {
 				logger.Log("err", err)
 				return
 			}
-			logger.Log("type", mt, "from", "server", "msg", string(msg))
+			logger.Log("from", "server", "msgId", msg.MsgId, "from", msg.MsgFrom, "To", msg.MsgTo, "content", msg.Content, "createdAt", msg.CreatedAt)
```
Comment (jarri-abidi) [inline cmd/client/main.go]: we can make this more readable this way:
```suggestion
			logger.Log(
			   "from", "server", 
			   "msgId", msg.MsgId, 
			   "from", msg.MsgFrom, 
			   "to", msg.MsgTo, 
			   "content", msg.Content, 
			   "createdAt", msg.CreatedAt,
			)
```

Comment (markhaur) [reply cmd/client/main.go]: cool

File: cmd/client/main.go [hunk 8]
```diff
@@ -11,11 +11,11 @@ import (
 )
 
 type Message struct {
-	MsgID     int64  `json: "message_id"`
-	MsgFrom   int64  `json: "message_from"`
-	MsgTo     int64  `json: "message_to"`
+	MsgID     int64  `json: "messageID"`
```
Comment (jarri-abidi) [inline cmd/client/main.go]: that's not camelCase, it's mixedCaps

File: cmd/server/chat/main.go [hunk 9]
```diff
@@ -9,11 +9,11 @@ import (
 )
 
 type Message struct {
-	MsgID     int64  `json:"message_id"`
-	MsgFrom   int64  `json: "message_from"`
-	MsgTo     int64  `json: "message_to"`
+	MsgID     int64  `json:"messageID"`
```
Comment (jarri-abidi) [inline cmd/server/chat/main.go]: same here

**Review by jarri-abidi — APPROVED** [review]

**Pull request merged**